collector/collector/report_handler.py

```diff
@@ -81,136 +110,196 @@ def before_request():
     )
 
 
+def validate_headers(headers, required_headers):
+    req_headers = dict(headers)
+    req_headers_keys = req_headers.keys()
+    for header in required_headers:
+        if header not in req_headers_keys:
+            raise InvalidUsage("Request header {} is missing".format(header), 400)
+        if not request.headers.get(header):
+            raise InvalidUsage("Request header {} is blank".format(header), 400)
+    return True
+
+
+def collector_post_handler():
+
+    tid_header = request.headers.get('X-Telemetry-TID')
+
+    # The collector only accepts records with the configured TID value.
+    # Make sure the TID in the collector config.py matches the TID
+    # configured for telemetrics-client on the systems from which this
+    # collector receives records.
+    if tid_header != TELEMETRY_ID:
+        err = "Telemetry ID mismatch. Expected: {}; Actual: {}".format(TELEMETRY_ID, tid_header)
+        raise InvalidUsage(err, 400)
+
+    record_format_version = request.headers.get('Record-Format-Version', -1)
+    if str(record_format_version).isdigit() and int(record_format_version) not in VALID_RECORD_FORMAT_VERSIONS:
+        raise InvalidUsage("Record-Format-Version is invalid", 400)
+
+    # Validate required headers based on Record Version
+    if int(record_format_version) == 1:
+        validate_headers(request.headers, REQUIRED_HEADERS_V1)
+    elif int(record_format_version) == 2:
+        validate_headers(request.headers, REQUIRED_HEADERS_V2)
+    elif int(record_format_version) == 3:
+        validate_headers(request.headers, REQUIRED_HEADERS_V3)
+    else:
+        raise InvalidUsage("Record-Format-Version is invalid", 400)
```

Comment:
This exception message is identical to the one on line 138. Can you change that one (or this one) to make them unique?

Reply:
Good point I'll update messages.